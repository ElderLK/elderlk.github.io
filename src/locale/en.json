{
  "header": {
    "link1": "Welcome",
    "link2": "Timeline",
    "link3": "Tooling",
    "link4": "Resume",
    "link5": "Contact"
  },
  "locale": {
    "en": "English",
    "pt": "Portuguese",
    "es": "Spanish"
  },
  "welcomeSection": {
    "title": "Welcome",
    "terminal": {
      "ms1": "Hi! üëãüèª",
      "ms2": "Welcome to my website.",
      "ms3": "I'm Elder, a software engineer rooted in Curitiba, Brazil.",
      "ms4": "Please feel free to explore this website and get to know more about me."
    }
  },
  "timelineSection": {
    "title": "My Brief timeline Overview",
    "1": {
      "date": "1992",
      "title": "Born",
      "description": "I was born on February 29, 1992, which was in a leap year.",
      "icon": "Baby"
    },
    "2": {
      "date": "2002",
      "title": "First computer",
      "description": "This is the year when I started loving computers, the year when my father brought a computer to our home.",
      "icon": "Computer"
    },
    "3": {
      "date": "2010",
      "title": "Teenage hood",
      "description": "I completed high school and a technical course in industrial mechanics.",
      "icon": "BookOpenCheck"
    },
    "4": {
      "date": "2012",
      "title": "Exploring a New Career Path",
      "description": "After working part-time in several small industries as a mechanic, I decided to change my career path and began my studies in Computer Science at UTFPR.",
      "icon": "Briefcase"
    },
    "5": {
      "date": "2016",
      "title": "A Fresh Start in a New Career",
      "description": "This was the year when I began my internship as a PHP developer at a logistics company, working on warehouse management software.",
      "icon": "Code2"
    },
    "6": {
      "date": "2017",
      "title": "Graduated",
      "description": "First, I graduated üéâ, and afterward, I was hired to work as a Salesforce developer in an Edtech company.",
      "icon": "GraduationCap"
    },
    "7": {
      "date": "2018",
      "title": "Transitioned to a New Job Role",
      "description": "I decided to apply for an open position within the company I was going to join the web team. I was accepted, and that's when I primarily began working with Node.js and TypeScript.",
      "icon": "ArrowLeftRight"
    },
    "8": {
      "date": "2020",
      "title": "Change Jobs",
      "description": "I moved to a new job where I face the challenge of creating a personal financial app from scratch, primarily using React and Node.",
      "icon": "Activity"
    },
    "9": {
      "date": "Nowadays",
      "title": "Engaging in Enhancing",
      "description": "I am presently working as a developer, creating and enhancing features in the financial app that I developed. Simultaneously, I am committed to ongoing self-study to further refine and expand my skill set.",
      "icon": "CalendarClock"
    }
  }
}
