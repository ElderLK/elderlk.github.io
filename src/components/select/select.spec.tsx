import React from 'react'
import { render, screen } from '@testing-library/react'
import { NextIntlClientProvider } from 'next-intl'
import { useRouter as mockUseRouter } from 'next-router-mock'

import { default as defaultLocale } from '../../locale/en.json'

import Select from './select.component'

jest.mock('next/navigation', () => ({
  useRouter: mockUseRouter,
  usePathname: jest.fn().mockReturnValue('/some-route')
}))

describe('<Select />', () => {
  it('should render the component', () => {
    const { container } = render(
      <NextIntlClientProvider locale="en" messages={defaultLocale}>
        <Select />
      </NextIntlClientProvider>
    )

    expect(screen.getByRole('button', { name: /English/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
