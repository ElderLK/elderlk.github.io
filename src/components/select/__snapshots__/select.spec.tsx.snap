// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render the component 1`] = `
<div
  class="relative"
>
  <button
    class="z-10 inline-flex flex-shrink-0 items-center rounded-md border border-gray-300 bg-gray-100 px-4 py-2.5 text-center text-sm font-medium text-gray-500 hover:bg-gray-200 focus:outline-none focus:ring-4 focus:ring-gray-100 dark:border-gray-600 dark:bg-gray-700 dark:text-white dark:hover:bg-gray-600 dark:focus:ring-gray-700"
    data-dropdown-toggle="dropdown-locales"
    id="locales-button"
    type="button"
  >
    <div
      class="h-4 w-4 text-2xl sm:h-6 sm:w-6"
    >
      <svg
        fill="none"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#us-flag)"
        >
          <path
            d="M256 512c141.385 0 256-114.615 256-256S397.385 0 256 0 0 114.615 0 256s114.615 256 256 256Z"
            fill="#F0F0F0"
          />
          <path
            d="M244.87 256.001H512c0-23.106-3.08-45.49-8.819-66.783H244.87v66.783ZM244.87 122.435h229.556a257.353 257.353 0 0 0-59.07-66.783H244.87v66.783ZM256 512.001c60.249 0 115.626-20.824 159.356-55.652H96.644c43.73 34.828 99.107 55.652 159.356 55.652ZM37.574 389.565h436.852a254.474 254.474 0 0 0 28.755-66.783H8.819a254.472 254.472 0 0 0 28.755 66.783Z"
            fill="#D80027"
          />
          <path
            d="M118.584 39.978h23.329l-21.7 15.765 8.289 25.509-21.699-15.765-21.699 15.765 7.16-22.037a257.407 257.407 0 0 0-49.652 55.337h7.475l-13.813 10.035a255.58 255.58 0 0 0-6.194 10.938l6.596 20.301-12.306-8.941a253.567 253.567 0 0 0-8.372 19.873l7.267 22.368h26.822l-21.7 15.765 8.289 25.509-21.699-15.765-12.998 9.444A258.468 258.468 0 0 0 0 256h256V0c-50.572 0-97.715 14.67-137.416 39.978Zm9.918 190.422-21.699-15.765L85.104 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509Zm-8.289-100.083 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765ZM220.328 230.4l-21.699-15.765L176.93 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509Zm-8.289-100.083 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765Zm0-74.574 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765Z"
            fill="#0052B4"
          />
        </g>
        <defs>
          <clippath
            id="us-flag"
          >
            <path
              d="M0 0h512v512H0z"
              fill="#fff"
            />
          </clippath>
        </defs>
      </svg>
    </div>
    <svg
      class="ml h-4 w-4 sm:h-6 sm:w-6"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m6 9 6 6 6-6"
      />
    </svg>
  </button>
  <div
    class="absolute top-0 z-10 w-44 divide-y divide-gray-100 rounded-lg bg-white  shadow ring-4 ring-gray-100 dark:bg-gray-700 dark:ring-gray-700 hidden"
    id="dropdown-locales"
  >
    <ul
      aria-labelledby="locales-button"
      class="py-2 text-sm text-gray-700 dark:text-gray-200"
    >
      <li>
        <button
          class="inline-flex w-full px-4 py-2 text-sm text-gray-700 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
          type="button"
        >
          <div
            class="inline-flex items-center"
          >
            <div
              class="h-4 w-4 text-2xl sm:h-6 sm:w-6"
            >
              <svg
                fill="none"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#us-flag)"
                >
                  <path
                    d="M256 512c141.385 0 256-114.615 256-256S397.385 0 256 0 0 114.615 0 256s114.615 256 256 256Z"
                    fill="#F0F0F0"
                  />
                  <path
                    d="M244.87 256.001H512c0-23.106-3.08-45.49-8.819-66.783H244.87v66.783ZM244.87 122.435h229.556a257.353 257.353 0 0 0-59.07-66.783H244.87v66.783ZM256 512.001c60.249 0 115.626-20.824 159.356-55.652H96.644c43.73 34.828 99.107 55.652 159.356 55.652ZM37.574 389.565h436.852a254.474 254.474 0 0 0 28.755-66.783H8.819a254.472 254.472 0 0 0 28.755 66.783Z"
                    fill="#D80027"
                  />
                  <path
                    d="M118.584 39.978h23.329l-21.7 15.765 8.289 25.509-21.699-15.765-21.699 15.765 7.16-22.037a257.407 257.407 0 0 0-49.652 55.337h7.475l-13.813 10.035a255.58 255.58 0 0 0-6.194 10.938l6.596 20.301-12.306-8.941a253.567 253.567 0 0 0-8.372 19.873l7.267 22.368h26.822l-21.7 15.765 8.289 25.509-21.699-15.765-12.998 9.444A258.468 258.468 0 0 0 0 256h256V0c-50.572 0-97.715 14.67-137.416 39.978Zm9.918 190.422-21.699-15.765L85.104 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509Zm-8.289-100.083 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765ZM220.328 230.4l-21.699-15.765L176.93 230.4l8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765 8.289 25.509Zm-8.289-100.083 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765Zm0-74.574 8.289 25.509-21.699-15.765-21.699 15.765 8.289-25.509-21.7-15.765h26.822l8.288-25.509 8.288 25.509h26.822l-21.7 15.765Z"
                    fill="#0052B4"
                  />
                </g>
                <defs>
                  <clippath
                    id="us-flag"
                  >
                    <path
                      d="M0 0h512v512H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <span
              class="ml-2"
            >
              English
            </span>
          </div>
        </button>
      </li>
      <li>
        <button
          class="inline-flex w-full px-4 py-2 text-sm text-gray-700 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
          type="button"
        >
          <div
            class="inline-flex items-center"
          >
            <div
              class="h-4 w-4 text-2xl sm:h-6 sm:w-6"
            >
              <svg
                fill="none"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#brazil-flag)"
                >
                  <path
                    d="M256 512c141.385 0 256-114.615 256-256S397.385 0 256 0 0 114.615 0 256s114.615 256 256 256Z"
                    fill="#6DA544"
                  />
                  <path
                    d="M256 100.174 467.478 256 256 411.826 44.522 256 256 100.174Z"
                    fill="#FFDA44"
                  />
                  <path
                    d="M256 345.043c49.177 0 89.043-39.866 89.043-89.043 0-49.177-39.866-89.043-89.043-89.043-49.177 0-89.043 39.866-89.043 89.043 0 49.177 39.866 89.043 89.043 89.043Z"
                    fill="#F0F0F0"
                  />
                  <path
                    d="M211.478 250.435c-15.484 0-30.427 2.355-44.493 6.725.623 48.64 40.227 87.884 89.015 87.884 30.168 0 56.812-15.017 72.919-37.968-27.557-34.497-69.958-56.641-117.441-56.641ZM343.393 273.06a89.45 89.45 0 0 0 1.651-17.06c0-49.178-39.866-89.043-89.043-89.043-36.694 0-68.194 22.201-81.826 53.899a183.693 183.693 0 0 1 37.305-3.812c51.717-.001 98.503 21.497 131.913 56.016Z"
                    fill="#0052B4"
                  />
                </g>
                <defs>
                  <clippath
                    id="brazil-flag"
                  >
                    <path
                      d="M0 0h512v512H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <span
              class="ml-2"
            >
              Portuguese
            </span>
          </div>
        </button>
      </li>
      <li>
        <button
          class="inline-flex w-full px-4 py-2 text-sm text-gray-700 hover:bg-gray-100 dark:text-gray-400 dark:hover:bg-gray-600 dark:hover:text-white"
          type="button"
        >
          <div
            class="inline-flex items-center"
          >
            <div
              class="h-4 w-4 text-2xl sm:h-6 sm:w-6"
            >
              <svg
                fill="none"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#spain-flag)"
                >
                  <path
                    d="M.001 256c0 31.314 5.633 61.31 15.923 89.043l240.077 22.261 240.077-22.261c10.29-27.733 15.923-57.729 15.923-89.043 0-31.314-5.633-61.31-15.923-89.043l-240.077-22.261-240.077 22.261C5.634 194.69.001 224.686.001 256Z"
                    fill="#FFDA44"
                  />
                  <path
                    d="M496.078 166.957C459.907 69.473 366.072 0 256.001 0S52.095 69.473 15.924 166.957h480.154ZM15.924 345.043C52.095 442.527 145.93 512 256.001 512s203.906-69.473 240.077-166.957H15.924Z"
                    fill="#D80027"
                  />
                </g>
                <defs>
                  <clippath
                    id="spain-flag"
                  >
                    <path
                      d="M.001 0h512v512h-512z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </div>
            <span
              class="ml-2"
            >
              Spanish
            </span>
          </div>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
